src/content/docs/identityserver/upgrades/identityserver4-to-duende-identityserver-v7.mdx

```diff
@@ -0,0 +1,487 @@
+    <Code
+      code={"dotnet list package | grep IdentityServer4"}
+      lang="bash"
+      title="Terminal"
+    />
+  </TabItem>
+</Tabs>
+
+This command will print a list of packages you are using in your solution, along with their version.
+
+```bash title="Output"
+   > IdentityServer4                            3.1.4       3.1.4
+   > IdentityServer4.EntityFramework            3.1.4       3.1.4
+```
+
+Depending on the package version shown, your next steps will be different:
+
+- If you are on IdentityServer v3.x, we recommend first [upgrading to IdentityServer4 v4.x](#identityserver4-v3x-to-identityserver-v4x), and then to [Duende IdentityServer](#identityserver4-v4x-to-duende-identityserver). The configuration object model changed between the two major versions of IdentityServer4, and we recommend upgrading step-by-step.
+- If you are on IdentityServer v4.x, you can immediately [upgrade to Duende IdentityServer](#identityserver4-v4x-to-duende-identityserver).
+
+## IdentityServer4 v3.x to IdentityServer v4.x
+
+The most straightforward upgrade path is to first update to the latest version of IdentityServer4 v4, and then continue to the latest Duende IdentityServer.
+
+Between IdentityServer4 v3.x and v4.x, the configuration object model was updated:
+
+- The relation between `ApiResources` and `ApiScopes` was changed from parent-child to many-to-many.
+- A number of configuration types were renamed:
+  - `ApiProperties` to `ApiResourceProperties`
+  - `ApiSecrets` to `ApiResourceSecrets`
+  - `IdentityClaims` to `IdentityResourceClaims`
+  - `IdentityProperties` to `IdentityResourceProperties`
+  - `ApiScopes` to `ApiResourceScopes`
+
+IdentityServer4 projects that use the `IdentityServer4.EntityFramework` package or implement their own stores will need to update their code and/or database to reflect these changes.
+
+Database changes will need to be done using a custom migration script, as using the default Entity Framework Core migration will result in data loss. We'll look at this later in this upgrade guide.
+
+:::tip[Sample project]
+We have a [sample project available on GitHub](https://github.com/DuendeSoftware/UpgradeSample-IdentityServer4-v3), which contains database migration scripts for these changes.
+:::
+
+### Step 1: Update NuGet Packages
+
+Update the IdentityServer4 dependencies in your IdentityServer host project to version `4.1.2`.
+
+```diff lang="xml" title=".csproj"
+- <PackageReference Include="IdentityServer4" Version="3.1.4" />
++ <PackageReference Include="IdentityServer4" Version="4.1.2" />
+```
+
+Make sure to change the version number for all IdentityServer4 packages, including `IdentityServer4.EntityFramework` and `IdentityServer4.AspNetIdentity`.
+
+### Step 2: Make Code Changes
+
+It's likely some code changes will be required, especially when using the IdentityServer UI templates.
+Visual Studio and JetBrains Rider provide great code completion that can be of help here.
```

Comment:
Do you want to link to the IDEs here? Linking out to credible sources helps SEO.